src/AutorestSwift/Code Generator/SwiftGenerator.swift

```diff
@@ -92,4 +105,20 @@ class SwiftGenerator: CodeGenerator {
             try object.toFile(inFolder: modelUrl)
         }
     }
+
+    private func generateChoice(for choice: ChoiceSchema) throws {
```

Comment:
So this will need to be a method to generate the enum file and will probably need to compose several templates.